lib/path.js

```diff
@@ -243,15 +243,21 @@ const parsePathData = (string) => {
 exports.parsePathData = parsePathData;
 
 /**
- * @type {(number: number, precision?: number) => string}
+ * @type {(number: number, precision?: number) => {
+ *   roundedStr: string,
+ *   rounded: number
+ * }}
  */
-const stringifyNumber = (number, precision) => {
+const roundAndStringify = (number, precision) => {
   if (precision != null) {
     const ratio = 10 ** precision;
     number = Math.round(number * ratio) / ratio;
   }
-  // remove zero whole from decimal number
-  return removeLeadingZero(number);
+
+  return {
```

Comment:
Wonder if this makes the RAM usage increase

Reply:
I'm not actually that experienced with memory profiling, but figured now's a good time to learn. I've taken a diff of two heap snapshots taken at the start and end of the `stringifyPathData` function.

I'm not sure what to think of `(object shape)` and `(compiled code)`, as these are managed by Node.js internally.

Excluding those:

* `(string)` - increase due to property names `rounded` and `roundedStr`.
* `Object`/`(array)`/`Array` - decrease since we scrapped the `combined` array.

Based on the objects that we can clearly trace to source code, we're using less memory. However, including `(object shape)` and `(compiled code)`, we're using more memory. I need to learn more about analyzing heap snapshots to know if further action should be taken on those.

## Before

![image](https://github.com/svg/svgo/assets/22801583/1bf72726-21b4-4de4-9665-7d57796b7f1d)

## After

![image](https://github.com/svg/svgo/assets/22801583/bb156536-cc23-4a6f-b6af-3db3c242d529)